### PR TITLE
Be able to use Shapash to compute Shapeley values through Shap for any model supported by Shap

### LINKED ISSUE #475: Support for LogisticRegression
Shapash is supposed to be compatible with LogisticRegression. However, we get the following error: 
```
from sklearn.linear_model import LogisticRegression
regressor = LogisticRegression().fit(Xtrain,ytrain)
xpl = SmartExplainer(
    model=regressor    
)
```
image

In addition, we need to review why this model is not tested in unit tests.

**Python version** :3.9

**Shapash version** :3.5.2

**Operating System** :Linux

### PR DESCRIPTION
**Pull request by guillaume-vignal**
# Description

This change will enable us to be able to use Shapash to compute Shapeley values through Shap for any model supported by Shap.

Fixes #475

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Test Configuration**:
* OS: linux
* Python version: 3.9
* Shapash version: 2.3.7

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by MLecardonnel — APPROVED** [review]

File: shapash/explainer/smart_explainer.py [hunk 1]
```diff
@@ -230,11 +228,13 @@ def __init__(
 
     def compile(self,
                 x,
+                backend=None,
```
Comment (ThomasBouche) [inline shapash/explainer/smart_explainer.py:231]: backend will be in init and compile ?

File: shapash/explainer/smart_explainer.py [hunk 2]
```diff
@@ -191,17 +190,16 @@ def __init__(
                 """
             )
         self.model = model
+        self.preprocessing = preprocessing
+        self.backend_name = None
```
Comment (ThomasBouche) [inline shapash/explainer/smart_explainer.py:194]: if you create an attribute can you add in docstrings attributes

Comment (guillaume-vignal) [reply shapash/explainer/smart_explainer.py:197]: done

File: shapash/explainer/smart_explainer.py (hunk 1 above)
Comment (guillaume-vignal) [reply shapash/explainer/smart_explainer.py]: finally I put it only in init

File: shapash/explainer/smart_explainer.py [hunk 3]
```diff
@@ -230,11 +228,13 @@ def __init__(
...
     def compile(self,
                 x,
+                backend=None,
                 contributions=None,
                 y_pred=None,
                 y_target=None,
                 additional_data=None,
-                additional_features_dict=None):
+                additional_features_dict=None,
+                **kwargs):
```
[the remark this answers is not in the document]
Comment (guillaume-vignal) [reply shapash/explainer/smart_explainer.py]: done

**Review by guerinclement — APPROVED** [review]

**Pull request merged**